### PR TITLE
IS-1120: prevalidate a tx more strictly 2

### PR DESCRIPTION
**Pull request by endlessgate**
## Pre-validation with strict for entered transaction request

# Update
* Version check
  - Support v3 protocol only
  - Field check
* Int-type check
  - Not allow uppercase character in hex string
  - Malformed number string: "0x03", "0x0003", "3"
* Address-type check
  - Malformed address format: "hx1234", "0xqw1234", "cxFEAC"
  - Invalid format check
* Required parameters check
* Pre-validate processing in icon-service

### CODE REVIEW

File: tests/unit_test/iconscore/test_icon_origin_pre_validator.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        ('', False),
+        ('0x1A', False),
+        ('0x3E', False),
+        ('0xBE', False),
+        ('0Xff', False),
+        ('0xff', True)
+    ]
+)
+def test_malformed_uppercase_string(validator, malformed_value, expected):
+    assert validator.is_integer_type(malformed_value) == expected
```
Comment (goldworm-icon) [inline tests/unit_test/iconscore/test_icon_origin_pre_validator.py:75]: Please test it with "0xzz" and "0x123k"?

File: tests/unit_test/iconscore/test_icon_origin_pre_validator.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    'malformed_value, expected',
+    [
+        ('', False),
+        ('0xqw1234', False),
+        ('hxEfe3', False),
+        ('hx1234', False),
+        ('cx1234', False),
+        ('cxEab1', False),
+        ('1234', False),
+        ('hx3f945d146a87552487ad70a050eebfa2564e8e5c', True)
```
Comment (goldworm-icon) [inline tests/unit_test/iconscore/test_icon_origin_pre_validator.py:88]: Add a normal contract address case

File: iconservice/iconscore/icon_pre_validator.py [hunk 3]
```diff
@@ -31,6 +32,23 @@
...
+    ConstantKeys.FROM,
+    ConstantKeys.TO,
+    ConstantKeys.SIGNATURE,
+    ConstantKeys.DATA_TYPE,
+    ConstantKeys.DATA,
+)
+
+REQUIRED_PARAMS = REQUEST_PARAMS[:4] + REQUEST_PARAMS[6:9]
+INT_PARAMS = REQUEST_PARAMS[:6]
+ADDR_PARAMS = REQUEST_PARAMS[6:8]
```
Comment (goldworm-icon) [inline iconservice/iconscore/icon_pre_validator.py:51]: Another way to improve code readability?

Comment (endlessgate) [reply iconservice/iconscore/icon_pre_validator.py:51]: What about change code like this?
```
REQUIRED_PARAMS = (
    ConstantKeys.VERSION,
    ConstantKeys.STEP_LIMIT,
    ConstantKeys.NID,
    ConstantKeys.TIMESTAMP,
    ConstantKeys.FROM,
    ConstantKeys.TO,
    ConstantKeys.SIGNATURE
)
INT_PARAMS = (
    ConstantKeys.VERSION,
    ConstantKeys.STEP_LIMIT,
    ConstantKeys.NID,
    ConstantKeys.TIMESTAMP,
    ConstantKeys.VALUE,
    ConstantKeys.NONCE
)
ADDR_PARAMS = (
    ConstantKeys.FROM,
    ConstantKeys.TO
)
```

**Pull request merged**